opus/application/static_media/js/hash.js

```diff
@@ -183,21 +319,121 @@ var o_hash = {
+                        extras[slug] = extras[slug] ? extras[slug] : [];
+                        while (extras[slug].length < slugCounter) {
+                            extras[slug].push(null);
+                        }
+                        extras[slug][slugCounter-1] = value;
+                    } else {
+                        // each qtype will only have one value at a time
+                        extras[slug] = [value];
+                    }
                 } else {
                     // Leave comments here, need to revisit this later. We have find
                     // a better way to tell if the slug value is coming from string input.
                     // if ($(`input[name="${slug}"]`).hasClass("STRING")) {
                     //     selections[slug] = [value];
                     // } else {
-                    selections[slug] = value.split(",");
+
+                    if (slugCounter) {
+                        slugCounter = parseInt(slugCounter);
+                        selections[slug] = selections[slug] ? selections[slug] : [];
+                        while (selections[slug].length < slugCounter) {
+                            selections[slug].push(null);
+                        }
+                        selections[slug][slugCounter-1] = value;
+                    } else {
+                        selections[slug] = value.split(",");
+                    }
                     // }
                 }
+            } else if (slug === "widgets" && value) {
+                // Loop through widgets and check if there is any widget with RANGE or STRING input
+                // but without qtype, we will put those in selections as well.
+                $.each(value.split(","), function(idx, widgetSlug) {
+                    if (`qtype-${widgetSlug}` in extras) {
+                        return;
+                    }
+
+                    let inputInAWidget = $(`#widget__${widgetSlug} input`);
+                    if (inputInAWidget.length === 0) {
+                        return;
+                    }
+
+                    // Sync up selections and opus.selections when a set of inputs is removed or added
+                    // so that page will not reload in these two cases.
+                    if (inputInAWidget.hasClass("RANGE")) {
+                        if (!(`${widgetSlug}1` in selections) && !(`${widgetSlug}2` in selections) &&
+                            !(`${widgetSlug}1` in opus.selections) && !(`${widgetSlug}2` in opus.selections)) {
+                            selections[`${widgetSlug}1`] = [null];
+                            selections[`${widgetSlug}2`] = [null];
+                            opus.selections[`${widgetSlug}1`] = [null];
+                            opus.selections[`${widgetSlug}2`] = [null];
+                        } else {
+                            selections[`${widgetSlug}1`] = (selections[`${widgetSlug}1`] ?
```

Comment:
Check out the JavaScript || and && operators. They do what's called "shortcutting". You should be able to say: `x = x || y`. If x evaluates to true, then x is assigned to x. If it evaluates to false, then x=y. So you can say:
```
selections[`${widgetSlug}2`] = selections[`${widgetSlug}2`] || [];
```
I think. This will make a lot of this code simpler. All see various functions later in this file for the same paradigm.